app/experimenter/experiments/tests/test_forms.py

```diff
@@ -832,85 +806,166 @@ def test_clean_unrecognized_countries(self):
         self.assertTrue(not form.is_valid())
         self.assertTrue(form.errors["countries"])
 
+    def test_form_is_invalid_if_population_percent_is_0(self):
+        self.data["population_percent"] = "0"
+        form = self.form_class(request=self.request, data=self.data)
+        self.assertFalse(form.is_valid())
+        self.assertIn("population_percent", form.errors)
 
-class TestExperimentVariantsPrefForm(MockRequestMixin, TestCase):
+    def test_form_is_invalid_if_population_percent_below_0(self):
+        self.data["population_percent"] = "-1"
+        form = self.form_class(request=self.request, data=self.data)
+        self.assertFalse(form.is_valid())
+        self.assertIn("population_percent", form.errors)
+
+    def test_form_is_invalid_if_population_percent_above_100(self):
+        self.data["population_percent"] = "101"
+        form = self.form_class(request=self.request, data=self.data)
+        self.assertFalse(form.is_valid())
+        self.assertIn("population_percent", form.errors)
+
+    def test_form_saves_population(self):
+        form = self.form_class(
+            request=self.request, data=self.data, instance=self.experiment
+        )
+
+        self.assertTrue(form.is_valid())
+
+        self.assertEqual(self.experiment.variants.count(), 0)
+
+        experiment = form.save()
+
+        self.assertEqual(
+            experiment.population_percent, decimal.Decimal("10.000")
+        )
+        self.assertEqual(
+            experiment.firefox_version, self.data["firefox_version"]
+        )
+        self.assertEqual(
+            experiment.firefox_channel, self.data["firefox_channel"]
+        )
+        self.assertEqual(
+            experiment.client_matching, self.data["client_matching"]
+        )
+
+
+class TestExperimentVariantsAddonForm(MockRequestMixin, TestCase):
 
     def setUp(self):
         super().setUp()
 
-        self.branch0_name = "control name"
-        self.branch1_name = "branch 1 name"
-        self.branch2_name = "branch 2 name"
-
         self.data = {
```

Comment:
Can we move all the common things from these now three massive dicts into just one. E.g.

```python
class TestExperimentVariantsBaseForm:
    FIXTURE_DATA = {
        "population_percent": "10",
        "firefox_version": Experiment.VERSION_CHOICES[-1][0],
        "firefox_channel": Experiment.CHANNEL_NIGHTLY,
        "client_matching": "en-us only please",
        "locales": [],
        "countries": [],
        ...
    }
    def setUp(self):
        self.data = copy.copy(self.FIXTURE_DATA)

class TestExperimentVariantsAddonForm:
    def setUp(self):
        self.data = copy.copy(TestExperimentVariantsBaseForm.FIXTURE_DATA)
        self.data.update({
            "addon_name": "Addon Name",
            "addon_experiment_id": "addon_experiment_id",
            "addon_testing_url": "https://www.example.com/testing.xpi",
            "addon_release_url": "https://www.example.com/release.xpi",
            ...anything else unique to this form class...
        })


class TestExperimentVariantsPrefForm:
    def setUp(self):
        self.data = copy.copy(TestExperimentVariantsBaseForm.FIXTURE_DATA)
        self.data.update({
            ...all things unique to this form class...
        })
```